lms/lmsdb/models.py

```diff
@@ -143,6 +153,44 @@ def random_password(cls, stronger: bool = False) -> str:
     def get_notifications(self) -> Iterable['Notification']:
         return Notification.fetch(self)
 
+    def notes_to_show(self) -> List[Dict[str, Any]]:
+        staff_roles = (
+            (current_user.role == Role.get_staff_role())
+            | (current_user.role == Role.get_admin_role())
+        )
+        private_notes = (
+            (Note.privacy == NoteOptions.PRIVATE.value)
+            & (Note.creator == current_user.id)
+        )
+        staff_notes = (Note.privacy == NoteOptions.STAFF.value) & (staff_roles)
+        user_notes = (
+            (Note.privacy == NoteOptions.USER.value)
+            & ((Note.user == current_user.id) | (staff_roles))
+        )
+        public_notes = Note.privacy == NoteOptions.PUBLIC.value
+
+        return list(Note.select().where(
```

Comment:
This query is SUPER complex, and you probably can easily simplify it using few simple Python condition.

Reply:
I think it's more complicated than that.
If you have an idea I'd like to hear.
Perhaps we can user List Comprehension, but it'll have same conditions as here.

Reply:
```python
public_or_mine = (
    Notes.privacy != NotePublicity.PRIVATE.value
    | Note.creator == current_user.id
)

notes = Notes.select().where(public_or_mine)

if not current_user.role.is_manager():
    notes.where(Note.privacy <= NotePublicity.USER.value)

return notes
```

Reply:
(In lmsdb/models.py, always prefer to return pure DB instances. It's better because the lazy loading of peewee hasn't tried to fetch anything yet).